actions/setup-nix/action.yml

```diff
@@ -0,0 +1,134 @@
+      required: false
+      default: ""
+
+  # github inputs ---------------------------------
+  github-token:
+    description: "GitHub auth token for private repos (if needed)"
+    required: false
+    default: ""
+
+  # grafana inputs --------------------------
+  metrics-job-name:
+    description: "grafana metrics job name"
+    required: false
+    default: manifest-build
+  gc-host:
+    description: "grafana hostname"
+    required: false
+  gc-basic-auth:
+    description: "grafana basic auth"
+    required: false
+  gc-org-id:
+    description: "grafana org/tenant id"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: configure AWS credentials
+      if: inputs.enable-aws == 'true'
+      uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        role-duration-seconds: ${{ inputs.role-duration-seconds }}
+        aws-region: ${{ inputs.aws-region }}
+    - name: setup AWS credentials for nix
+      if: inputs.enable-aws == 'true'
+      shell: bash
+      run: |
+        # set up folder with AWS configs
+        aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
+        aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
+        aws configure set aws_session_token $AWS_SESSION_TOKEN
+        aws configure set region ${{ inputs.aws-region }}
+
+        # determine root path
+        ROOT_PATH=""
+        if [[ "${{ runner.os }}" == "Linux" ]]; then
+          ROOT_PATH="/root"
+        elif [[ "${{ runner.os }}" == "macOS" ]]; then
+          ROOT_PATH="/var/root"
+        else
+          echo "${{ runner.os }} not supported"
+          exit 1
+        fi
+
+        # symlink credential folders for nix build access
+        sudo ln -s ~/.aws $ROOT_PATH
+
+    - uses: cachix/install-nix-action@6004951b182f8860210c8d6f0d808ec5b1a33d28 # v25
+      with:
+        install_url: https://releases.nixos.org/nix/nix-2.18.1/install
+        nix_path: nixpkgs=channel:nixos-unstable
+        # Enables sandbox for all drv except those with `__noChroot = true;`
+        # Add caches for public nixos cache + public community cache + specified private cache
+        extra_nix_config: |
+          sandbox = relaxed
+          substituters = https://cache.nixos.org https://nix-community.cachix.org ${{ inputs.cache-url }}
+          trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs= ${{ inputs.cache-pubkey }}
+
+    - uses: cachix/cachix-action@18cf96c7c98e048e10a83abd92116114cd8504be # v14
+      if: inputs.enable-cachix == 'true'
+      with:
+        name: ${{ inputs.cachix-name }}
+        # If you chose API tokens for write access OR if you have a private cache
+        authToken: '${{ inputs.cachix-token }}'
+
+    - uses: MrSquaare/ssh-setup-action@7bfdda6ef6839504e3a5b2625ad971b17461a0db # v3
```

Comment:
nix also supports SSH substituters so having SSH setup here seemed useful since this action combines the functionality of adding various caches
https://nixos.org/manual/nix/stable/package-management/ssh-substituter